app/uk/gov/hmrc/cbcr/repositories/ReportingEntityDataRepo.scala

```diff
@@ -116,33 +120,67 @@ class ReportingEntityDataRepo @Inject()(protected val mongo: ReactiveMongoApi)(i
     )
   }
 
-  def queryCbcId(cbcId: CBCId, reportingPeriod: LocalDate) : Future[Option[ReportingEntityData]] = {
+  def queryCbcId(cbcId: CBCId, reportingPeriod: LocalDate): Future[Option[ReportingEntityData]] = {
     val criteria = Json.obj(
-      "reportingEntityDRI" -> Json.obj("$regex" -> (".*" + cbcId.toString + ".*" )),
-        "reportingPeriod" -> reportingPeriod.toString
+      "reportingEntityDRI" -> Json.obj("$regex" -> (".*" + cbcId.toString + ".*")),
+      "reportingPeriod" -> reportingPeriod.toString
     )
 
     Logger.info(s"ReportingEntityData retrieval query criteria: $criteria")
     repository.flatMap(_.find(criteria).one[ReportingEntityData])
   }
 
-  def query(c:String, r:String) : Future[Option[ReportingEntityData]] = {
+  def queryTIN(tin: String): Future[List[ReportingEntityData]] = {
+    val criteria = Json.obj("tin" -> tin)
+
+    val result: Future[List[ReportingEntityData]] = repository.flatMap(_.find(criteria)
+      .cursor[ReportingEntityData]()
+      .collect[List](-1, Cursor.FailOnError[List[ReportingEntityData]]()))
+
+
+    result.map(x => getLatestReportingEntityData(x))
+
+  }
+
+  def getLatestReportingEntityData(reportingEntityData: List[ReportingEntityData]) = {
+    val timestampRegex = """\d{8}T\d{6}""".r
+
+    implicit val localDateTimeOrdering = new Ordering[LocalDateTime] {
+      def compare(x: LocalDateTime, y: LocalDateTime): Int = {
+        x.compareTo(y)
+      }
+    }
+
+    val reportingEntityDri: Seq[String] = reportingEntityData.map(data => data.reportingEntityDRI.id)
+
+    val timestamps: Seq[LocalDateTime] = reportingEntityDri.map { dri =>
+      timestampRegex.findFirstIn(dri) match {
+        case Some(data) => LocalDateTime.parse(data, DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss"))
+      }
+    }
+
+    val timestampSeparator = Set('-', ':')
```

Comment:
In future we can refactor this to use headOption but based on current scenarios the timestamps will always have a value if is reaching that endpoint due to previous validations